src/types.ts

```diff
@@ -104,9 +105,18 @@ export interface Context<T = any> {
    */
   contextId: number
 
+  /**
+   * Describes the language processor to be used for evaluation
+   */
   executionMethod: ExecutionMethod
 
   evaluationMethod: EvaluationMethod
+
+  /**
+   * Describes the strategy / paradigm to be used for evaluation
+   * Examples: lazy, concurrent or non-deterministic
+   */
+  variant?: Variant // This will replace evaluationMethod
```

Comment:
Would it be better if `variant` were a required property? 
Since:
- there is a default value for it ('default')
- the default value is set when creating the context if no value for `variant` is specified. Therefore, there isn't a case where the context is created without having the property

Reply:
Yes, I think it can be a required property. I've changed it.